File: code_stubs/sort.java
import java.util.*;

public class ArraySorter {

    public static void printArray(int[] arr) {
        System.out.println("{");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]);
            if (i != arr.length - 1) {
                System.out.print(" ");
            }
        }
        System.out.println("\n}");
    }

    public static void sort(int[] arr) {
        // Your code goes here!

        printArray(arr);  // Make sure to call this print after every iteration
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter numbers separated by spaces: ");
        String input = scanner.nextLine();

        String[] inputStrings = input.split(" ");
        int[] arr = new int[inputStrings.length];

        for (int i = 0; i < inputStrings.length; i++) {
            arr[i] = Integer.parseInt(inputStrings[i]);
        }

        printArray(arr);
        sort(arr);
        printArray(arr);
    }
}

File: code_stubs/sort.rs
use std::io;

fn print_array(arr: &[i32]) {
    println!("{{");
    for (i, &num) in arr.iter().enumerate() {
        if i == arr.len() - 1 {
            print!("{}", num);
        } else {
            print!("{} ", num);
        }
    }
    println!("\n}}");
}

fn sort(arr: &mut Vec<i32>) {
    // Your code goes here!

    print_array(arr);  // Make sure to call this print after every iteration
}

fn main() {
    println!("Enter numbers separated by spaces: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let arr: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Parse error"))
        .collect();

    print_array(&arr);
    let mut sortable_arr = arr.clone();
    sort(&mut sortable_arr);
    print_array(&sortable_arr);
}
